src/main/java/io/github/dsheirer/filter/FilterEditorPanel.java

```diff
@@ -98,7 +98,7 @@ private void addFilter(Filter<T> filter, DefaultMutableTreeNode parent)
     {
         List<FilterElement<?>> elements = filter.getFilterElements();
 
-        elements.sort(Comparator.comparing(FilterElement::toString));
+        elements.sort(Comparator.comparing(FilterElement::getName));
```

Comment:
order filter elements by their display name (as above) instead of synthetic `toString`